src/Game/LevelController.cpp

```diff
@@ -109,18 +109,34 @@ void LevelController::draw_editor()
 {
     static i32 minutes = 0;
     static i32 seconds = 0;
-    bool is_changed = false;
+    bool is_time_changed = false;
 
     ImGui::Text("Map Time");
-    is_changed |= ImGui::InputInt("Minutes: ", &minutes);
-    is_changed |= ImGui::InputInt("Seconds: ", &seconds);
+    is_time_changed |= ImGui::InputInt("Minutes: ", &minutes);
+    is_time_changed |= ImGui::InputInt("Seconds: ", &seconds);
 
-    if (is_changed)
+    if (is_time_changed)
     {
         map_time = minutes * 60 + seconds;
     }
-
     ImGui::Text(("Time: " + std::to_string(time)).c_str());
+
+    ImGui::Separator();
+
+    static i32 new_map_food = 0;
+    bool is_food_changed = false;
+
+    ImGui::Text("Map Food");
+    is_food_changed |= ImGui::InputInt("Food: ", &new_map_food);
+
+    if (is_food_changed)
+    {
+        map_food = new_map_food;
+    }
+    ImGui::Text(("Food: " + std::to_string(Player::get_instance()->food) + " / " + std::to_string(map_food)).c_str());
```

Comment:
I think you could just write
```c++
ImGui::InputInt("Food: ", &map_food);
```
ImGui "accepts" input when you go out of focus, meaning that if you just go with &map_food it will be updated only when you change the value.

Reply:
Unfortunately map_food is u32 variable and ImGui don't take reference u32 as an argument in InputInt. Maybe there is a better way to do it. But I can remove the is_food_changed variable.

Reply:
ImGui methods that take other types such as u32 were already mentioned on Discord, you can use InputScalar: https://github.com/ocornut/imgui/issues/643#issuecomment-386720788